pgjdbc/src/test/java/software/aws/rds/jdbc/postgresql/ca/AwsDriverTest.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (c) 2004, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package software.aws.rds.jdbc.postgresql.ca;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import software.aws.rds.jdbc.postgresql.Driver;
+
+import org.junit.Before;
+import org.junit.jupiter.api.Test;
+import org.postgresql.PGProperty;
+import org.postgresql.test.TestUtil;
+import org.postgresql.util.LogWriterHandler;
+import org.postgresql.util.NullOutputStream;
+import org.postgresql.util.URLCoder;
+
+import java.io.PrintWriter;
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Properties;
+import java.util.logging.Handler;
+import java.util.logging.Logger;
+
+/*
+ * Tests the dynamically created class software.aws.rds.jdbc.postgresql.Driver
+ *
+ */
+public class AwsDriverTest {
+
+  @Before
+  public void setUp() throws SQLException {
+
+    if (!software.aws.rds.jdbc.postgresql.Driver.isRegistered()) {
+
+      software.aws.rds.jdbc.postgresql.Driver.register();
+
+    }
+  }
+
+  @Test
+  public void urlIsNotForPostgreSQL() throws SQLException {
+    Driver driver = new Driver();
+
+    assertNull(driver.connect("jdbc:otherdb:database", new Properties()));
+  }
+
+  /**
+   * According to the javadoc of java.sql.Driver.connect(...), calling abort when the {@code executor} is {@code null}
+   * results in SQLException
+   */
+  @Test
+  public void urlIsNull() {
+    Driver driver = new Driver();
+
+    assertThrows(SQLException.class, () -> driver.connect(null, new Properties()));
+  }
+
+  @Test
+  public void testAcceptAwsProtocolOnly() throws Exception {
+    Driver drv = new Driver();
+    assertNotNull(drv);
+
+    verifyUrl(drv, "jdbc:postgresql:aws://localhost/test", "localhost", "5432", "test");
+    assertFalse(drv.acceptsURL("jdbc:postgresql://localhost:5432/test"));
+  }
+
+  /*
+   * This tests the acceptsURL() method with a couple of well and poorly formed jdbc urls.
+   */
+  @Test
+  public void testAcceptsURLAws() throws Exception {
```

Comment:
Do we need test that check that community protocol isn't supported by AWS driver?